### DISCUSSION TITLE
Datoformater - hvor går grensen?

### DISCUSSION THREAD

**Discussion by frisol97** [fremtind/jokul#1725] [Annet]
Siden vi nå har startet et arbeid rundt validering av inputfeltet til datovelgeren blir det naturlig å diskutere hvilke formater vi skal godta og ikke. Slik den fungerer nå vil ikke noe skje før brukeren har fylt inn en gyldig dato i formatet dd.mm.åååå. 

Spørsmålet er om det er tilstrekkelig å begrense det til disse formatene eller om det er behov for flere?

**Forslag om gyldige datoformater:**
dd.mm.åååå
dd-mm-åååå
dd/mm/åååå

Med tekst
dd.månedtekst

Relatert til #1605

**Comment by vegardsoerlie** [comment]
Flott, vi diskuterte også dette i går på teamet. I dagens løsning konverterer vi `, / -` til `.`. Skulle gjerne ønske at det var funksjonalitet som lever videre. Men det er neppe behov for at bruker trenger å se at tegnet konverteres slik vi har det i dag, der er jeg usikker på hva som blir best.

**Reply by frisol97** [reply]
Men dere har ingen behov for at det skulle være åååå.dd.mm for eksempel?

**Reply by vegardsoerlie** [reply]
Nei, kun ddmmåååå

**Comment by joms** [comment]
Jeg ser ikke noen gode grunner til å støtte flere formater enn det vi har i dag. Ett format gjør det enklere om man navigerer mellom flere Fremtind-løsninger, og inputfeltet oppfører seg likt over alt.

Hvis det skulle vært noe er det å ha mulighet til å droppe årstall jeg ser for meg kunne vært nyttig for pro-tjenester hvor det skrives mye input raskt.

Normalisering av datoformatet tenker jeg må være opp til hver enkelt tjeneste i bakkant, og det bør ikke være opp til brukeren å støtte opp under all verdens formater. F.eks. om en tjeneste lagrer tidsdata i Unix Epoch, eller om de bruker ISO 8601, må det være opp til tjenesten selv å normalisere dataene.

**Comment by Saegrov** [comment]
Jeg er enig med @joms - jeg ser ingen grunn til at vi skal støtte noen andre datoformater enn det som brukes på Norsk. I vår løsning konverterer vi alle datoer til ISO8601 når vi lagrer det i databaser, men når man skal skrive input så syns jeg at norsk datoformat holder.

**Comment by frisol97** [comment]
Konklusjonen er da altså at det ikke er behov for å støtte noe annet enn norsk datoformat med dato-måned-år. 
Og selve konverteringen trenger ikke å vises for brukeren i inputfeltet, men gjøres i bakgrunnen.

**Comment by piofinn** [comment]
Hvis vi kun godtar norsk datoformat, og ikke gjør omformatering, har vi fortsatt et problem med informasjon til brukeren. Det bør være en skikkelig tilbakemelding på om (og i så fall hvorfor) det som er skrevet inn er ugyldig, med mindre man gjør omformatering (normalisering) i feltet under skriving.

Utfordringen her er at uansett løsning beveger man seg inn på området skjemahåndtering, som jo løses av klient. Jeg tenker kanskje dermed at man bør se nærmere på APIet til komponenten og se om man klarer å legge bedre til rette for normalisering og gode feilmeldinger på klientside. Kanskje prøve å nærme oss APIet til standard `<input type="date">`?

**Reply by lfbergee** [reply]
Det er vel ikke vanlig at man kan velge sitt eget datoformat i en datovelger? Er ikke noe av poenget med en felleskomponent for dette å velge en standard?

Normalt bruker man en default dato eller placeholder til å vise hva som forventes, og da er det vel ikke logisk at du bare kan hive inn det du har lyst til? Hva som sendes inn til baksystemene er jo helt irrelevant for frontend komponenten, det gir bare tilbake sin verdi også kan du gjøre hva du vil med den etter det.
